File: 2022/LeetCode/problem58.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function(s) {
    var count = 0;
    var start = 1;

    while (s[s.length - count - start] === " ") {
        start++;
    }

    while (s.length - count - start >= 0 && s[s.length - count - start] !== " ") {
        count++;
    }

    return count;
    
    /*
    var wordArr = s.trim().split(" ")

    var lastWord = wordArr[wordArr.length - 1]

    return lastWord.length
    */
};

File: 2022/Leetcode/problem58.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        if (s.length() == 1) {
            return 1;
        }
        
        int end = s.length() - 1;
        while (end >= 0 && !isalpha(s[end])) {
            end--;
        }

        int start = end - 1;  
        while (start >= 0 && isalpha(s[start])) {
            start--;
        }
        return end - start;
    }
};